pkg/receive/handler_test.go

```diff
@@ -120,3 +128,346 @@ func TestCountCause(t *testing.T) {
+			wreq:              wreq1,
+			appendables: []*fakeAppendable{
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, nil, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, nil, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, nil, nil),
+				},
+			},
+		},
+		{
+			name:              "size 3 commit error",
+			status:            http.StatusInternalServerError,
+			replicationFactor: 1,
+			wreq:              wreq1,
+			appendables: []*fakeAppendable{
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, commitErrFn, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, commitErrFn, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, commitErrFn, nil),
+				},
+			},
+		},
+		{
+			name:              "size 3 commit error with replication",
+			status:            http.StatusInternalServerError,
+			replicationFactor: 3,
+			wreq:              wreq1,
+			appendables: []*fakeAppendable{
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, commitErrFn, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, commitErrFn, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(nil, nil, commitErrFn, nil),
+				},
+			},
+		},
+		{
+			name:              "size 3 appender error with replication",
+			status:            http.StatusInternalServerError,
+			replicationFactor: 3,
+			wreq:              wreq1,
+			appendables: []*fakeAppendable{
+				&fakeAppendable{
+					appender:    newFakeAppender(nil, nil, nil, nil),
+					appenderErr: appenderErrFn,
+				},
+				&fakeAppendable{
+					appender:    newFakeAppender(nil, nil, nil, nil),
+					appenderErr: appenderErrFn,
+				},
+				&fakeAppendable{
+					appender:    newFakeAppender(nil, nil, nil, nil),
+					appenderErr: appenderErrFn,
+				},
+			},
+		},
+		{
+			name:              "size 3 conflict with replication",
+			status:            http.StatusConflict,
+			replicationFactor: 3,
+			wreq:              wreq1,
+			appendables: []*fakeAppendable{
+				&fakeAppendable{
+					appender: newFakeAppender(conflictErrFn, nil, nil, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(conflictErrFn, nil, nil, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(conflictErrFn, nil, nil, nil),
+				},
+			},
+		},
+		{
+			name:              "size 3 conflict and commit error with replication",
+			status:            http.StatusConflict,
+			replicationFactor: 3,
+			wreq:              wreq1,
+			appendables: []*fakeAppendable{
+				&fakeAppendable{
+					appender: newFakeAppender(conflictErrFn, nil, commitErrFn, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(conflictErrFn, nil, commitErrFn, nil),
+				},
+				&fakeAppendable{
+					appender: newFakeAppender(conflictErrFn, nil, commitErrFn, nil),
+				},
+			},
+		},
+		{
+			name:              "size 3 with replication and one faulty",
```

Comment:
I think I'd be worth it to add a few more cases where 2/3 appenders fail with an error.

Reply:
ack, added more tests for 1/3 faulty receivers and 2/3 faulty receivers with different mixes of errors, which indeed are some of the more interesting cases.

Reply:
Nice :+1: